orchagent/bulker.h

```diff
@@ -0,0 +1,656 @@
+#pragma once
+
+#include <assert.h>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+#include <stdexcept>
+#include <boost/functional/hash.hpp>
+#include <sairedis.h>
+#include "sai.h"
+#include "logger.h"
+
+static inline bool operator==(const sai_ip_prefix_t& a, const sai_ip_prefix_t& b)
```

Comment:
could you not put code in header files ? i guess since inline, then you are trying to gain speed here ? probably not much gain, i think would be better if you put this code in cpp for better readability

Reply:
@qiluo-msft I agree with Kamil here. I see that there are a lot of template functions in this file, but have you considered doing something like this: https://isocpp.org/wiki/faq/templates#separate-template-fn-defn-from-decl

Reply:
@daall There is no template function, only template struct or classes.

---
In reply to: [414741883](https://github.com/Azure/sonic-swss/pull/1238#discussion_r414741883) [](ancestors = 414741883)